packages/wrangler/src/__tests__/metrics.test.ts

```diff
@@ -438,13 +449,47 @@ describe("metrics", () => {
 function mockUserRequest() {
 	const requests = { count: 0 };
 	beforeEach(() => {
-		setMockResponse("/user", () => {
-			requests.count++;
-			return { id: "MOCK_USER_ID" };
-		});
+		msw.use(
+			...mswSuccessOauthHandlers,
+			rest.get("*/user", (_, res, cxt) => {
+				requests.count++;
+				return res(
+					cxt.status(200),
+					cxt.json({
+						success: true,
+						errors: [],
+						messages: [],
+						result: { id: "MOCK_USER_ID" },
+					})
+				);
+			})
+		);
 	});
 	afterEach(() => {
 		requests.count = 0;
 	});
 	return requests;
 }
+
+function mockMetricRequest(
+	body: unknown,
+	header: unknown,
+	endpoint: "identify" | "event"
+) {
+	const requests = { count: 0 };
+	msw.use(
+		rest.post(`*/${endpoint}`, async (req, res, cxt) => {
+			requests.count++;
+			expect(await req.json()).toEqual(body);
+			expect(req.headers).toContain(header);
+			return res.once(cxt.status(200), cxt.json({}));
+		})
+	);
+
+	return requests;
+}
+
+// Forces a tick to allow the non-awaited fetch promise to resolve.
+function flushPromises(): Promise<void> {
+	return new Promise((resolve) => setImmediate(resolve));
+}
```

Comment:
Before anyone asks, this is another example of behaviors being mocked away in tests. 
In this case the fetch is not awaited (purposefully to prevent blocking behavior) this means the tests need a moment to tick over so that the `Promise` can resolve, this edge behavior was never captured in the previous tests.